ckan/lib/dictization/__init__.py

```diff
@@ -132,7 +131,11 @@ def table_dict_save(table_dict, ModelClass, context):
             params = dict((key, table_dict.get(key)) for key in constraint)
             obj = session.query(ModelClass).filter_by(**params).first()
             if obj:
-                break
+                if 'name' in params and getattr(obj, 'state', None) == State.DELETED:
+                    obj.name = obj.id
+                    obj = None
+                else:
+                    break
```

Comment:
Oh, I'm wrong. this code is trying to prevent the postgres error, so it's actually fine. :+1:

At a later time we can update this code to rely on postgres enforcement instead of avoid it, but that's out of scope for this change.